aiohomekit/controller/ble/key.py

```diff
@@ -45,3 +45,15 @@ def decrypt(self, data: bytes | bytearray):
         data = self.key.decrypt(b"", counter, bytes([0, 0, 0, 0]), data)
         self.counter += 1
         return data
+
+
+class BroadcastDecryptionKey:
+    def __init__(self, key: bytes) -> None:
+        self.key = ChaCha20Poly1305Decryptor(key)
+        self.counter = 0
+
+    def decrypt(self, data: bytes | bytearray):
+        counter = self.counter.to_bytes(8, byteorder="little")
```

Comment:
Is counter always 0 for this key?

Reply:
I think so but the docs are actually cut off in the PDF file

Reply:
LOL

Reply:
So the AAD is the advertising ID

and the Nonce is the Global state number

Now we have to figure out the advertising id

Reply:
Its just the id 

So the next issue is the auth tag is truncated to 4 bytes instead of 16 bytes so it won't decrypt.

32bittruncatedAuthTag.

Reply:
Can probably use this to only compare the first 4 bytes of the tag

https://github.com/ph4r05/py-chacha20poly1305/blob/master/chacha20poly1305/__init__.py